scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/SimulationFacade.java

```diff
@@ -71,6 +71,8 @@
   //simulation results from the last simulation, as output directly by simulation driver
   private SimulationResults lastSimDriverResults;
 
+  private Map<String, ActivityInstanceId> activityNameToActivityId;
```

Comment:
I'd unify this declaration with its assignment (on line 184), that way you can sidestep any possible nullability runtime exceptions. Line 184 could be changed to `activityNameToActivityId.clear()` if this map is expected to be populated multiple times.

Reply:
You're right, I have done that.